### ISSUE TITLE
[Bug] Shell's GoToAsync doesn't update top tabbar's bold tab (iOS only)

### ISSUE THREAD

**Issue by LaloCo** [xamarin/Xamarin.Forms#6362]
### Description
On iOS, when using GoToAsync to navigate from one -top- tab to another, the top TabBar still marks the origin tab as bold, but the navigation does happen. The white "underline" also changes to the destination tab, so only the bold stays in the origin tab. Android does behave as expected.

### Steps to Reproduce

1. Have two top tabs
```xml
<ShellItem Route="Third"
           Title="Third element"
           Icon="main.png">
    <ShellSection Route="Section">
        <ShellContent Route="Main"
                      Title="Main"
                      Icon="main.png">
            <local:MainPage/>
        </ShellContent>
        <ShellContent Route="Secondary"
                      Title="Secondary"
                      Icon="sec.png">
            <local:SecondaryPage/>
        </ShellContent>
    </ShellSection>
</ShellItem>
```
2. On the first (Third/Section/Main) tab, add a button and have it execute this on click:
```csharp
Shell.Current.GoToAsync("//Third/Section/Secondary", true);
```

### Expected Behavior
1. The navigation should happen
2. The underline should change to the destination tab
3. The destination tab's title should be set to bold.

### Actual Behavior
1. The navigation does happen
2. The underline does change to the destination tab
3. The destination tab's title is not set to bold.

### Basic Information

- Version with issue: 4.0.0.425677
- Last known good version: --
- IDE: Visual Studio for Mac
- Platform Target Frameworks: 
  - iOS:  12.2
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Screenshots

![Simulator Screen Shot - iPhone Xʀ - 2019-05-30 at 10 50 46](https://user-images.githubusercontent.com/4640571/58645361-d596a300-82c8-11e9-83e4-04ffdf3adc44.png)

**Comment by samhouts** [comment]
@LaloCo  Can you please attach a small project that demonstrates this issue? Thanks!

**Comment by LaloCo** [comment]
Hey @samhouts, sure thing, [this repo](https://github.com/LaloCo/ShellExample) shows it. It has a "Third item" flyout item which has the two top tabs. Clicking on the Secondary tab works as expected, but clicking the button on the first tab creates the issue described.

**Comment by jfversluis** [comment]
On Android it does work as it should, on iOS the selected tab visualisation does not update when navigating outside of tapping the actual tabbar.

**Comment by samhouts** [comment]
closed by #8599
